src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -7141,13 +7141,16 @@ int RGWRados::Object::Read::read(int64_t ofs, int64_t end,
 
   state.cur_ioctx->locator_set_key(read_obj.loc);
 
-  // set localise read flag if the option is set
-  int localise_read_flag = 0;
-  if (store->ctx()->_conf.get_val<bool>("rgw_localize_read")) {
-    localise_read_flag = librados::OPERATION_LOCALIZE_READS;
+  // set read flag if the option is set
+  int read_flag = 0;
+  auto read_policy = store->ctx()->_conf.get_val<std::string>("rgw_replica_read_policy");
+  if (read_policy == "localize") {
```

Comment:
I don't think this should happen here. This should be part of the RGWRados state (e.g., keep `read_flag` there), or somewhere where it's not checked every time.

Reply:
Thanks for the suggestion will work in the next iteration for the PR